File: frontend/src/entities/purchase-order/model/purchase-order-status.ts
/**
 * Purchase order status constants and configuration.
 * Labels are handled via i18n (purchasing.json purchaseOrder.status section).
 */

export const PurchaseOrderStatus = {
  DRAFT: 'DRAFT',
  SENT: 'SENT',
  CONFIRMED: 'CONFIRMED',
  RECEIVED: 'RECEIVED',
  CANCELED: 'CANCELED',
} as const;

export type PurchaseOrderStatus = typeof PurchaseOrderStatus[keyof typeof PurchaseOrderStatus];

export interface StatusConfig {
  readonly color: 'steel' | 'info' | 'success' | 'warning' | 'danger';
}

/**
 * Display configuration for each purchase order status.
 * Colors map to design system Badge variants.
 * Labels are in locales/{lang}/purchasing.json under "purchaseOrder.status" key.
 */
export const PurchaseOrderStatusConfig: Record<PurchaseOrderStatus, StatusConfig> = {
  [PurchaseOrderStatus.DRAFT]: { color: 'steel' },
  [PurchaseOrderStatus.SENT]: { color: 'info' },
  [PurchaseOrderStatus.CONFIRMED]: { color: 'warning' },
  [PurchaseOrderStatus.RECEIVED]: { color: 'success' },
  [PurchaseOrderStatus.CANCELED]: { color: 'danger' },
};

File: backend/src/main/java/com/wellkorea/backend/purchasing/domain/vo/PurchaseOrderStatus.java
package com.wellkorea.backend.purchasing.domain.vo;

/**
 * Status enum for PurchaseOrder lifecycle.
 * DRAFT → SENT → CONFIRMED → RECEIVED, or CANCELED at any point before RECEIVED.
 */
public enum PurchaseOrderStatus {
    /**
     * PO created but not sent.
     */
    DRAFT,

    /**
     * PO sent to vendor.
     */
    SENT,

    /**
     * Vendor confirmed the order.
     */
    CONFIRMED,

    /**
     * Goods/services received.
     */
    RECEIVED,

    /**
     * Order canceled.
     */
    CANCELED
}
